addons/mail/static/src/core/common/thread_service.js

```diff
@@ -749,6 +727,10 @@ export class ThreadService {
         }
         const message = this.store.Message.insert(data, { html: true });
         thread.messages.add(message);
+        if (thread.selfMember && !message.isSeenBySelf) {
```

Comment:
(note) condition on master is slightly different

Reply:
This condition is slightly better as it rely on an existing field, can be updated on master as well.